src/zenml/services/local/local_service.py

```diff
@@ -0,0 +1,335 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+import os
+import pathlib
+import subprocess
+import sys
+import tempfile
+from abc import abstractmethod
+from typing import Dict, List, Optional, Tuple
+
+import psutil
+from pydantic import Field
+
+from zenml.logger import get_logger
+from zenml.services.local.local_service_endpoint import (
+    LocalDaemonServiceEndpoint,
+)
+from zenml.services.service import BaseService, ServiceConfig
+from zenml.services.service_status import ServiceState, ServiceStatus
+
+logger = get_logger(__name__)
+
+
+SERVICE_DAEMON_CONFIG_FILE_NAME = "service.json"
+SERVICE_DAEMON_LOG_FILE_NAME = "service.log"
+SERVICE_DAEMON_PID_FILE_NAME = "service.pid"
+
+
+class LocalDaemonServiceConfig(ServiceConfig):
+    """Local daemon service configuration.
+
+    Attributes:
+        silent_daemon: set to True to suppress the output of the daemon
+            (i.e. redirect stdout and stderr to /dev/null). If False, the
+            daemon output will be redirected to a logfile.
+    """
+
+    silent_daemon: bool = False
+
+
+class LocalDaemonServiceStatus(ServiceStatus):
+    """Local daemon service status.
+
+    Attributes:
+        runtime_path: the path where the service daemon runtime files (the
+            configuration file used to start the service daemon and the
+            logfile) are located
+        silent_daemon: flag indicating whether the output of the daemon
+            is suppressed.
+    """
+
+    runtime_path: Optional[str] = None
+    silent_daemon: bool = False
+
+    @property
+    def config_file(self) -> str:
```

Comment:
I think these should be `Optional[str]` instead of returning an empty string (same for remaining properties)